chapter05/dcox/ex02.pl

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/perl
+use strict;
+use warnings;
+
+my %total_bytes;
+my %source_bytes;
+while ( <> ) {
+  my ( $source, $destination, $bytes ) = split;
+  $total_bytes{$source}{$destination} += $bytes;
+  $source_bytes{$source} += $bytes;
+}
+
+for my $source ( sort { $source_bytes{$b} <=> $source_bytes{$a} } keys %total_bytes ) {
+  my $source_total = 0;
+  for my $destination ( keys %{ $total_bytes{$source} } ) {
+    $source_total += $total_bytes{$source}{$destination};
+  }
+  print "$source - $source_total\n";
```

Comment:
I think you're right. I'm sure I added this before I added the sorting, then didn't realise I could simplify it.